store/sqlstore/shared_channel_store.go

```diff
@@ -216,8 +216,16 @@ func (s SqlSharedChannelStore) getSharedChannelsQuery(opts model.SharedChannelFi
 		Select(selectStr).
 		From("SharedChannels AS sc")
 
+	if opts.MemberId != "" {
+		query = query.Join("ChannelMembers AS cm ON cm.ChannelId = sc.ChannelId").
+			Where(sq.Eq{"cm.UserId": opts.MemberId})
+	}
+
 	if opts.TeamId != "" {
-		query = query.Where(sq.Eq{"sc.TeamId": opts.TeamId})
+		query = query.Where(sq.Or{
+			sq.Eq{"sc.TeamId": opts.TeamId},
+			sq.Eq{"sc.TeamId": ""},
```

Comment:
Why was the empty TeamId condition added here? Asking because on first looks it doesn't seem directly related to the issue the PR is addressing.

Reply:
While I was looking at how best to check membership in the sqlstore, I saw this code that lead me to believe teamId="" is a valid case:
https://github.com/mattermost/mattermost-server/blob/8a7af3bef29d7acedc31851794d406bf46371b0e/store/sqlstore/channel_store.go#L960

I agree it is unrelated.

Reply:
DMs/GMs have empty team ids. But here we are inside a check on `opts.TeamId != ""` so it's a bit weird to then include empty fields. Anyway, not a blocker just something I've noticed.

Reply:
Good point.  I'll remove it.